tf2onnx/tfonnx.py

```diff
@@ -250,11 +250,18 @@ def arg_minmax_op(ctx, node, name, args):
 
 def reduce_op(ctx, node, name, args):
     axes_node = node.inputs[1]
-    input_rank = len(ctx.get_shape(node.input[0]))
     axes = axes_node.get_tensor_value()
     if np.isscalar(axes):
         axes = [axes]
-    axes = [val + input_rank if val < 0 else val for val in axes]
+    input_shape = ctx.get_shape(node.input[0])
+    if input_shape is None:
+        for val in axes:
```

Comment:
maybe use any([val < 0 for val in axes])

Reply:
sold